service/service.go

```diff
@@ -513,39 +513,51 @@ func (s *Service) GetToken(ctx context.Context, req *pbsvc.UserRequest) (*pbsvc.
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
 
-	// TODO include mapping table in db
-	permissionLevel := auth.PermissionEnumMap[retrievedUser.GetPermissionLevel()]
+	var identity *pblib.Identification
 
-	// build token header, body, secret
-	header := &auth.Header{
-		Alg:      auth.AlgorithmMap[permissionLevel],
-		TokenTyp: auth.Jwt,
-	}
-	body := &auth.Body{
-		UUID:                retrievedUser.GetUuid(),
-		Permission:          permissionLevel,
-		ExpirationTimestamp: time.Now().UTC().Add(time.Hour * time.Duration(jwtExpirationTime)).Unix(),
-	}
+	existingToken, err := getExistingToken(retrievedUser.GetUuid())
+	if err != nil || (existingToken != nil && existingToken.permission != retrievedUser.GetPermissionLevel()) {
+		// TODO include mapping table in db
+		permissionLevel := auth.PermissionEnumMap[retrievedUser.GetPermissionLevel()]
```

Comment:
Your algorithm should be:
`if existingToken.permission != retrievedUser.GetPermissionLevel()`, then do not proceed further as discussed from the comment above.

`if err != nil`, then log  the error, and make a new token.